esmvaltool/diag_scripts/galytska23/select_variables_for_tigramite.py

```diff
@@ -124,55 +146,50 @@ def calculate_heat_flux(list_va_ta):
-        out_var = calculate_slp(item)
-        out_var.var_name = 'Psl_Ural'
-    elif var == 'pressure_sib':
-        out_var = calculate_slp(item)
-        out_var.var_name = 'Psl_Sib'
-    elif var == 'pressure_aleut':
-        out_var = calculate_slp(item)
-        out_var.var_name = 'Psl_Aleut'
-    elif var == 'bk_ice':
-        out_var = finalize_bk_ice(item)
-    elif var == 'ok_ice':
-        out_var = finalize_ok_ice(item)
-    elif var == 'heat_flux':
-        out_var = prepare_heat_flux(item)
+    if var_name == 'pv':
+        out_var = calculate_polar_vortex(var)
+    elif var_name == 'pre_tas':
+        out_var = calculate_arctic_tas(var)
+    elif var_name in [
+            'pressure_ural',
+            'pressure_sib',
+            'pressure_aleut',
+    ]:
+        out_var = calculate_slp(var)
+    elif var_name == 'bk_ice':
+        out_var = finalize_bk_ice(var)
+    elif var_name == 'ok_ice':
+        out_var = finalize_ok_ice(var)
+    elif var_name == 'heat_flux':
+        out_var = prepare_heat_flux(var)
     else:
-        raise NotImplementedError(f"Variable '{var}' not supported")
+        raise NotImplementedError(f"Variable '{var_name}' not yet supported.")
     return out_var
 
 
-def calculate_variables(input_dict):
+def calculate_variables(dataset_dict):
     """Calculate all necessary variables."""
     logger.debug("Variables are calculated for the following datasources:%s",
-                 input_dict.keys())
-    dictionary = {}
-    for key, value in input_dict.items():
-        logger.debug("Calculating final variables for %s dataset", key)
-        dictionary.setdefault(key, {})
-        tmp_list = []
-        for item in value:
-            if item['preprocessor'] == "heat_flux":
-                tmp_list.append(variable_cases(item['preprocessor'], item))
-            else:
-                dictionary[key].setdefault(
-                    variable_cases(item['preprocessor'], item).var_name,
-                    variable_cases(item['preprocessor'], item)
-                )
+                 dataset_dict.keys())
+    processed_vars = {}
+    for dataset, variables in dataset_dict.items():
+        processed_vars[dataset] = {}
 
-        if key != "HadISST":
-            # Calculate heat flux for all data sources except HadISST
-            heat_flux = calculate_heat_flux(tmp_list)
-            dictionary[key].setdefault(heat_flux.var_name, heat_flux)
-    return dictionary
+        logger.debug("Calculating final variables %s for %s dataset",
+                     variables, dataset)
+
+        for var in variables:
+            var_name = var['preprocessor']
+            # get the list of variables to be ignored per dataset
+            to_ignore_vars = ignored_variables.get(dataset, None)
+            if var_name not in to_ignore_vars:
+                new_var = variable_cases(var_name, var)
+                new_var_name = new_var.var_name
+                processed_vars[dataset][new_var_name] = new_var
```

Comment:
So there are two major issues here: 
1. Suggested `if var_name not in to_ignore_vars:` fails because it retuns `None ` for the models that are not in `ignored_variables`. So the script crashes. I tried to make a work around with
 `if to_ignore_vars is not None and var_name not in to_ignore_vars or to_ignore_vars is None:` (worked just fine) when I realized that the new version does not include  `calculate_heat_flux `function. 
3. So I suggest the code below (however, probably is not the most elegant solution) that includes `calculate_heat_flux `. I also added the error message `raise IndexError("The preprocessor heat flux requests two variables in the recipe: va and ta")`, since `tmp_list` should include in the recipe two variables (**va** and **ta**) for each dataset in order to calculate the heat flux, I realized that it could easily be a common pitfall. 
```suggestion
        if dataset in ignored_variables:
            to_ignore_vars = ignored_variables.get(dataset, None)
            for var in variables:
                var_name = var['preprocessor']
                if var_name not in to_ignore_vars:
                    new_var = variable_cases(var_name, var)
                    new_var_name = new_var.var_name
                    processed_vars[dataset][new_var_name] = new_var
        else: 
            tmp_list = []
            for var in variables:
                var_name = var['preprocessor']
                if var_name == "heat_flux":
                    tmp_list.append(variable_cases(var_name, var))
                else: 
                    new_var = variable_cases(var_name, var)
                    new_var_name = new_var.var_name
                    processed_vars[dataset][new_var_name] = new_var
            if len (tmp_list) !=2:
                raise IndexError("The preprocessor heat flux requests two variables in the recipe: va and ta")
            heat_flux = calculate_heat_flux(tmp_list)
            processed_vars[dataset][heat_flux.var_name] = heat_flux
```